src/Avalonia.Controls/Shapes/Shape.cs

```diff
@@ -248,60 +247,29 @@ protected void InvalidateGeometry()
 
         protected override Size MeasureOverride(Size availableSize)
         {
-            bool deferCalculateTransform;
-            switch (Stretch)
+            if (DefiningGeometry is null)
             {
-                case Stretch.Fill:
-                case Stretch.UniformToFill:
-                    deferCalculateTransform = double.IsInfinity(availableSize.Width) || double.IsInfinity(availableSize.Height);
-                    break;
-                case Stretch.Uniform:
-                    deferCalculateTransform = double.IsInfinity(availableSize.Width) && double.IsInfinity(availableSize.Height);
-                    break;
-                case Stretch.None:
-                default:
-                    deferCalculateTransform = false;
-                    break;
+                return default;
             }
 
-            if (deferCalculateTransform)
-            {
-                _calculateTransformOnArrange = true;
-                return DefiningGeometry?.Bounds.Size ?? Size.Empty;
-            }
-            else
-            {
-                _calculateTransformOnArrange = false;
-                return CalculateShapeSizeAndSetTransform(availableSize);
-            }
+            return CalculateSizeAndTransform(availableSize, DefiningGeometry.Bounds, Stretch).Item1;
```

Comment:
Perhaps adding names to tuple members will make this a bit nicer `return (...).size` and not `Item1`?

Reply:
Yeah probably, though it's an internal method that's defined just below that methods that call it so I wasn't too bothered by the results being unnamed. I'll add the names in though.